packages/nextjs/src/client/index.tsx

```diff
@@ -5,29 +5,64 @@ import React from 'react';
 
 export * from '@clerk/clerk-react';
 
-const NO_FRONTEND_API_ERR =
-  'The NEXT_PUBLIC_CLERK_FRONTEND_API environment variable must be set to use the ClerkProvider component.';
+//const NO_FRONTEND_API_ERR = 'The NEXT_PUBLIC_CLERK_FRONTEND_API environment variable must be set to use the ClerkProvider component.';
 
-type NextClerkProviderProps = {
-  children: React.ReactNode;
-  frontendApi?: string;
-} & IsomorphicClerkOptions;
+type NextClerkProviderProps =
+  | ({
+      children: React.ReactNode;
+      publishableKey?: string;
+      frontendApi: undefined;
+    } & IsomorphicClerkOptions)
+  | ({
+      children: React.ReactNode;
+      publishableKey: undefined;
+      frontendApi?: string;
+    } & IsomorphicClerkOptions);
+
+// TODO: Undup
+function parsePublishableKey(key: string, pkg: string) {
```

Comment:
(TODO):

Drop the `shared` internal "package" and replace it with a shared package that is actually reusable. Can probably move errors there as well

Reply:
Done